kernels/portable/cpu/op_grid_sampler_2d.cpp

```diff
@@ -0,0 +1,257 @@
+
+            // Get corner pixel values with boundary checking
+            T nw = get_pixel_value(
+                input_data, n, c, iy_nw, ix_nw, inp_H, inp_W, padding_mode);
+            T ne = get_pixel_value(
+                input_data, n, c, iy_nw, ix_nw + 1, inp_H, inp_W, padding_mode);
+            T sw = get_pixel_value(
+                input_data, n, c, iy_nw + 1, ix_nw, inp_H, inp_W, padding_mode);
+            T se = get_pixel_value(
+                input_data,
+                n,
+                c,
+                iy_nw + 1,
+                ix_nw + 1,
+                inp_H,
+                inp_W,
+                padding_mode);
+
+            // Perform bilinear interpolation (weighted sum)
+            out_data[out_idx] =
+                static_cast<T>(nw * w_nw + ne * w_ne + sw * w_sw + se * w_se);
+          } else if (interpolation_mode == 1) { // Nearest
+            // Convert to pixel space and round to nearest pixel
+            float ix = unnormalize_coord(x, inp_W, align_corners);
+            float iy = unnormalize_coord(y, inp_H, align_corners);
+
+            int64_t ix_nearest = static_cast<int64_t>(std::round(ix));
+            int64_t iy_nearest = static_cast<int64_t>(std::round(iy));
+
+            // Get nearest pixel value
+            out_data[out_idx] = get_pixel_value(
+                input_data,
+                n,
+                c,
+                iy_nearest,
+                ix_nearest,
+                inp_H,
+                inp_W,
+                padding_mode);
+          }
+        }
+      }
+    }
+  }
+}
+
+// Main grid_sampler_2d function that validates inputs and dispatches to
+// implementation
+Tensor& grid_sampler_2d_out(
+    KernelRuntimeContext& ctx,
+    const Tensor& input,
+    const Tensor& grid,
+    int64_t interpolation_mode,
+    int64_t padding_mode,
+    bool align_corners,
+    Tensor& out) {
+  const int64_t N = input.size(0);
+  const int64_t C = input.size(1);
+  const int64_t out_H = grid.size(1);
+  const int64_t out_W = grid.size(2);
+
+  // Check for 4D input and grid
+  ET_KERNEL_CHECK(ctx, (input.dim() == 4), InvalidArgument, out);
+  ET_KERNEL_CHECK(ctx, (grid.dim() == 4), InvalidArgument, out);
+  ET_KERNEL_CHECK(ctx, (grid.size(3) == 2), InvalidArgument, out);
+
+  // Check that grid is float type
+  ET_KERNEL_CHECK(
+      ctx, (grid.scalar_type() == ScalarType::Float), InvalidArgument, out);
+
+  // Check interpolation mode is valid (0=bilinear, 1=nearest, 2=bicubic)
+  // 2=bicubic unavailable
+  ET_KERNEL_CHECK(
+      ctx,
+      (interpolation_mode >= 0 && interpolation_mode < 2),
+      InvalidArgument,
+      out);
+
+  // Check padding mode is valid (0=zeros, 1=border, 2=reflection)
+  ET_KERNEL_CHECK(
+      ctx, (padding_mode >= 0 && padding_mode <= 2), InvalidArgument, out);
+
+  // Check for output shape
+  ET_KERNEL_CHECK(
+      ctx,
+      (out.size(0) == N && out.size(1) == C && out.size(2) == out_H &&
+       out.size(3) == out_W),
```

Comment:
shouldn't we be attempting to resize output to the desired size so that we can handle dynamic shapes smoothly? @manuelcandales

Reply:
Yes.
@DenisVieriu97 please follow this [pattern](https://outlook.office.com/mail/inbox/id/AAQkAGZiZWI2YTcxLTg5YzYtNDY4MS05YzhkLTEyNjkzYjdjOTc5OAAQABmXiRYUZchEgiT0bC1lF%2B8%3D)